lib/Machine.build.js

```diff
@@ -116,6 +116,35 @@ module.exports = function Machine_build (machineDefinition) {
         }
       }
 
+      // If a `type` is given, validate that it's a valid type,
+      // and either provide an example or, if there's already an
+      // example, ensure that the example is of the given type.
+      if (input.type) {
+
+        // Ensure the type is valid.
+        if (!_.contains(['string', 'number', 'boolean', 'json', 'ref'], input.type)) {
+          throw flaverr('MACHINE_INPUT_INVALID', new Error('Failed to instantiate machine "' + machineDefinition.friendlyName + '" from the specified machine definition.\nThe machine\'s `'+inputKey+'` input has an invalid `type` ("' + input.type + '").  Type must be one of "string", "number", "boolean", "json" or "ref".'));
+        }
+
+        // If no example was given, use the default exemplar for the given type.
+        if (!input.example) {
```

Comment:
I think you want:

```javascript
if (!_.isUndefined(input.example)) {
```


_(`false`, empty string, and 0 are all valid exemplars)_